.github/workflows/python-CI.yaml

```diff
@@ -0,0 +1,42 @@
+name: Python CI
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    paths:
+      - "python/**"
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+concurrency:
+  group: test-python-${{ github.head_ref }}
+  cancel-in-progress: true
+
+defaults:
+  run:
+    working-directory: ./python
+
+jobs:
+  ci:
+    name: CI Python
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+      - name: Set up Python 3.8
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.8"
+      - name: Install tox
+        run: pip install tox==4.11.4
+      - name: Check style
+        run: |
+          tox run -e ruff
+          git diff --exit-code
+      - name: Run mypy
+        run: |
+          tox run -e mypy-openai
```

Comment:
Yes. theirs is very sophisticated, because their repo is very complicated, but we can actually do this much more simply. It's not showing now, but tox itself will manage that for us pretty easily.